pytential/symbolic/pde/system_utils.py

```diff
@@ -0,0 +1,529 @@
+        result += int_g.copy(source_kernels=(knl,), target_kernel=target_kernel,
+                densities=(density * c,), kernel_arguments=new_kernel_args)
+    return result
+
+
+def get_deriv_relation(kernels: List[ExpressionKernel],
+        base_kernel: ExpressionKernel,
+        kernel_arguments: Mapping[Text, Any],
+        tol: float = 1e-10,
+        order: Optional[int] = None) \
+        -> List[Tuple[ExpressionT, List[Tuple[Tuple[int], ExpressionT]]]]:
+    res = []
+    for knl in kernels:
+        res.append(get_deriv_relation_kernel(knl, base_kernel,
+            hashable_kernel_arguments=hashable_kernel_args(kernel_arguments),
+            tol=tol, order=order))
+    return res
+
+
+@memoize_on_first_arg
+def get_deriv_relation_kernel(kernel: ExpressionKernel,
+        base_kernel: ExpressionKernel,
+        hashable_kernel_arguments: Tuple[Tuple[Text, Any]],
+        tol: float = 1e-10,
+        order: Optional[int] = None) \
+        -> Tuple[ExpressionT, List[Tuple[Tuple[int], ExpressionT]]]:
+    """Takes a *kernel* and a base_kernel* as input and re-writes the
+    *kernel* as a linear combination of derivatives of *base_kernel* up-to
+    order *order* and a constant. *tol* is an upper limit for small numbers that
+    are replaced with zero in the numerical procedure.
+
+    Returns the constant and a list of (mulit-index, coeff) to represent the
+    linear combination of derivatives
+    """
+    kernel_arguments = dict(hashable_kernel_arguments)
+    (L, U, perm), rand, mis = _get_base_kernel_matrix(base_kernel, order=order,
+            hashable_kernel_arguments=hashable_kernel_arguments)
+    dim = base_kernel.dim
+    sym_vec = make_sym_vector("d", dim)
+    sympy_conv = SympyToPymbolicMapper()
+
+    expr = _get_kernel_expression(kernel.expression, kernel_arguments)
+    vec = []
+    for i in range(len(mis)):
+        vec.append(evalf(expr.xreplace(dict((k, v) for
+            k, v in zip(sym_vec, rand[:, i])))))
+    vec = sym.Matrix(vec)
+    result = []
+    const = 0
+    logger.debug("%s = ", kernel)
+
+    sol = lu_solve_with_expand(L, U, perm, vec)
+    for i, coeff in enumerate(sol):
+        coeff = chop(coeff, tol)
+        if coeff == 0:
+            continue
+        if mis[i] != (-1, -1, -1):
+            coeff *= _get_kernel_expression(kernel.global_scaling_const,
+                    kernel_arguments)
+            coeff /= _get_kernel_expression(base_kernel.global_scaling_const,
+                    kernel_arguments)
+            result.append((mis[i], sympy_conv(coeff)))
+            logger.debug("  + %s.diff(%s)*%s", base_kernel, mis[i], coeff)
+        else:
+            const = sympy_conv(coeff * _get_kernel_expression(
+                kernel.global_scaling_const, kernel_arguments))
+    logger.debug("  + %s", const)
+    return (const, result)
+
+
+@memoize_on_first_arg
+def _get_base_kernel_matrix(base_kernel: ExpressionKernel,
+        hashable_kernel_arguments: Tuple[Tuple[Text, Any]],
```

Comment:
```suggestion
        hashable_kernel_arguments: Tuple[Tuple[Text, Any], ...],
```

Reply:
Done in 36d81452ebfdfd726fdf7bec95d5de4710c1f906